conans/test/functional/tools/test_pkg_config.py

```diff
@@ -0,0 +1,61 @@
+import os
+
+import pytest
+
+from conan.tools.gnu.pkgconfig import PkgConfig
+from conans.errors import ConanException
+from conans.model.new_build_info import NewCppInfo
+from conans.test.utils.mocks import ConanFileMock
+from conans.test.utils.test_files import temp_folder
+from conans.util.files import save
+
+libastral_pc = """
+PC FILE EXAMPLE:
+
+prefix=/usr/local
+exec_prefix=${prefix}
+libdir=${exec_prefix}/lib
+includedir=${prefix}/include
+
+Name: libastral
+Description: Interface library for Astral data flows
+Version: 6.6.6
+Libs: -L${libdir}/libastral -lastral -Wl,--whole-archive
+Cflags: -I${includedir}/libastral -D_USE_LIBASTRAL
+"""
+
+
+@pytest.mark.tool_pkg_config
+class TestPkgConfig:
+    def test_negative(self):
+        conanfile = ConanFileMock()
+        pkg_config = PkgConfig(conanfile, 'libsomething_that_does_not_exist_in_the_world')
+        with pytest.raises(ConanException):
+            pkg_config.libs()
+
+    def test_pc(self):
+        tmp_dir = temp_folder()
+        filename = os.path.join(tmp_dir, 'libastral.pc')
+        save(filename, libastral_pc)
+
+        conanfile = ConanFileMock()
+        pkg_config = PkgConfig(conanfile, "libastral", pkg_config_path=tmp_dir)
+
+        assert pkg_config.provides == "libastral = 6.6.6"
+        assert pkg_config.version == "6.6.6"
+        assert pkg_config.includedirs == ['/usr/local/include/libastral']
+        assert pkg_config.defines == ['-D_USE_LIBASTRAL']
+        assert pkg_config.libs == ['astral']
+        assert pkg_config.libdirs == ['/usr/local/lib/libastral']
+        assert pkg_config.linkflags == ['-Wl,--whole-archive']
+        assert pkg_config.variables['prefix'] == '/usr/local'
+
+        cpp_info = NewCppInfo()
+        pkg_config.cpp_info(cpp_info)
+
+        assert cpp_info.includedirs == ['/usr/local/include/libastral']
+        assert cpp_info.defines == ['-D_USE_LIBASTRAL']
+        assert cpp_info.system_libs == ['astral']
```

Comment:
The `is_system` argument is a good starter, but feels a bit too too "hard".
Perhaps the method can accept a list argument to differentiate between ordinary and system libraries?
e.g. if a lib is in `["dl", "m", "pthread", "rt"]` then it is a system library, else it is an ordinary lib.

This would require the logic to have access to the current `settings.os`.

Reply:
Why would need access to ``settings.os``? The new approach has it, because it has access to ``self._conanfile``, as every other new ``conan.tool.xxx``, but not fully clear how the ``settings.os`` could help here.

The argument for filtering some system libs sounds good, I'll try to add it.

Reply:
e.g. to add `["user32", "shlwapi", "ws2_32", ...]` to system_libs on Windows and `["dl", "m", "pthread", "rt", ...]` on Linux systems.
Maintaining such a system might become a nightmare though.

Reply:
But isn't it just a filter? I have proposed ``pkg_config.fill_cpp_info(cpp_info, system_libs=["user32", "ws2_32", "m", "pthread", "rt", ...]``. I'd prefer not to have this encoded in Conan, we are trying to move away of this kind of hardcoded knowledge of data in helpers, and most of these things should be provided by recipes or conf. 

Because if we do it in Conan, it is always an infinite number of tickets, requesting: "hey, this system library "xxx" is not being correctly assigned to ``system_libs`` and it is returned as ``lib``. And if we add that library, maybe for another user that library will be in a Conan package because they are not depending on the system one, but on the Conan package one, and then we have a blocker.

Reply:
Completely agree. A filter is a good middle ground.